File: backend/BCAS.Api/Model/Announcement.cs
namespace BCAS.Api.Model;

public class Announcement
{
    public int Id { get; set; }
    public string Title { get; set; } = string.Empty;
    public string Body { get; set; } = string.Empty;
    public int? DepartmentId { get; set; }
    public string? DepartmentName { get; set; }
    public string Status { get; set; } = string.Empty;
    public DateTime? PublishAtUtc { get; set; }
    public int CreatedBy { get; set; }
    public string? CreatedByName { get; set; }
    public DateTime CreatedAt { get; set; }
    public int? UpdatedBy { get; set; }
    public string? UpdatedByName { get; set; }
    public DateTime? UpdatedAt { get; set; }
}

File: frontend/src/types/announcement.ts
import type { ContentStatusValue } from './news';

export interface AnnouncementDto {
  id: number;
  title: string;
  body: string;
  departmentId: number | null;
  departmentName: string | null;
  status: ContentStatusValue;
  effectiveDateUtc: string | null;
  createdByName: string | null;
  createdAt: string;
  updatedByName: string | null;
  updatedAt: string | null;
}

export interface AnnouncementRequest {
  title: string;
  body: string;
  effectiveDateUtc: string;
  status: ContentStatusValue;
}
